src/sentry_envelope.c

```diff
@@ -59,11 +68,16 @@ sentry__prepared_http_request_free(sentry_prepared_http_request_t *req)
 static sentry_envelope_item_t *
 envelope_add_item(sentry_envelope_t *envelope)
 {
-    if (envelope->item_count >= MAX_ENVELOPE_ITEMS) {
+    if (envelope->is_raw) {
+        return NULL;
```

Comment:
_nit_: Should we add a warning message here, since this indicates an implementation bug?

Reply:
true. Lets keep it in mind. But in theory users should deal with `NULL` already.